[skip ci][Maps] Update search document section with new features (#44819)
elastic/kibana: docs/maps/search.asciidoc

[[maps-search]]
== Searching your data

**Elastic Maps** embeds the search bar for real-time search.
Only layers requesting data from {es} are filtered when you submit a search request.

> Layers narrowed by the search context contain the icon image:maps/images/filter_icon.png[] to the right of layer name in the legend.

Reviewer: Can we give this icon a name?  So it reads something like "contain the tool icon"

** Grid aggregation source

** <<terms-join>>. The search context is applied to both the terms join and the vector source when the vector source is provided by Elasticsearch documents.

* <<heatmap-layer>> with Grid aggregation source

[role="screenshot"]
image::maps/images/global_search_bar.png[]

[role="xpack"]
[[maps-create-filter-from-map]]
=== Creating filters from your map

You can create filters by interacting with your map.

> Filters are applied to all layers in your map and other panels when your map is <<maps-embedding, embedded in a dashboard>>

Author: I replace this section with just links to the types of filters that can be created

> 

Reviewer: How about adding something like this:

*Elastic Maps* supports two types of interactive filters:

- Spatial filters
- Phrase filters

and link down the page.

[float]
[[maps-create-spatial-filters]]

> ==== Create spatial filters

Reviewer: ```suggestion
==== Spatial filters
```

Spatial filters narrow search results to documents that either intersect, are within, or do not intersect with the specified geometry.

You can create spatial filters by:

* Clicking image:maps/images/tools_icon.png[] to define the spatial fitler by drawing a polygon or a bounding box on the map.
* Clicking *Filter by geometry* in a feature tooltip to use a feature's geometry for the spatial filter.
+
[role="screenshot"]
image::maps/images/create_spatial_filter.png[]

You can configure the following spatial filter properties:

* Use *Geometry label* to provide a distinguishable and meaningful name for your spatial filter.
* *Spatial field* specifies the geo_point or geo_shape field used to determine if a document matches the spatial relation with the specified geometry.
* *Spatial relation* determines the {ref}/query-dsl-geo-shape-query.html#_spatial_relations.html[spatial relation operator] used at search time. Spatial relation selection is only available when the selected Spatial field is geo_shape.

[float]
[[maps-create-phrase-filter]]

> ==== Create phrase filters

Reviewer: ```suggestion
==== Phrase filters
```

** Grid aggregation source

** <<terms-join>>. The search context is applied to both the terms join and the vector source when the vector source is provided by Elasticsearch documents.

* <<heatmap-layer>> with Grid aggregation source

[role="screenshot"]
image::maps/images/global_search_bar.png[]

[role="xpack"]
[[maps-create-filter-from-map]]
=== Creating filters from your map

You can create two types of filters by interacting with your map:

* <<maps-spatial-filters, Spatial filters>>
* <<maps-phrase-filter, Phrase filters>>

[float]
[[maps-spatial-filters]]
==== Spatial filters

> Spatial filters narrow search results to documents that either intersect, are within, or do not intersect with the specified geometry.

Reviewer: ```suggestion
A spatial filter narrows search results to documents that either intersect with, are within, or do not intersect with the specified geometry.
```

You can create spatial filters in two ways:

> * Clicking image:maps/images/tools_icon.png[] to define the spatial fitler by drawing a polygon or a bounding box on the map.

Reviewer: * Click the tool icon image:maps/images/tools_icon.png[], and then draw a polygon or bounding box on the map to define the spatial filter.

Searching across multiple indices might sometimes result in empty layers.
The most common cause for empty layers are searches for a field that exists in one index, but does not exist in other indices.

[float]
[[maps-disable-search-for-layer]]
==== Disable search for layer

> You can disable the globar search bar from applying search context to a layer by unchecking *Apply global filter to layer* in Layer settings.

Reviewer: ```suggestion
To prevent the global search bar from applying search context to a layer, clear the *Apply global filter to layer* checkbox in Layer settings.
```
